src/OpenTelemetry.AutoInstrumentation/NativeCallTargetDefinition.cs

```diff
@@ -0,0 +1,142 @@
+// <copyright file="NativeCallTargetDefinition.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace OpenTelemetry.AutoInstrumentation;
+
+ // !                                         ██
+ //                                         ██░░██
+ //                                       ██░░░░░░██
+ //                                     ██░░░░░░░░░░██
+ //                                     ██░░░░░░░░░░██
+ //                                   ██░░░░░░░░░░░░░░██
+ //                                 ██░░░░░░██████░░░░░░██
+ //                                 ██░░░░░░██████░░░░░░██
+ //                               ██░░░░░░░░██████░░░░░░░░██
+ //                               ██░░░░░░░░██████░░░░░░░░██
+ //                             ██░░░░░░░░░░██████░░░░░░░░░░██
+ //                           ██░░░░░░░░░░░░██████░░░░░░░░░░░░██
+ //                           ██░░░░░░░░░░░░██████░░░░░░░░░░░░██
+ //                         ██░░░░░░░░░░░░░░██████░░░░░░░░░░░░░░██
+ //                         ██░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░██
+ //                       ██░░░░░░░░░░░░░░░░██████░░░░░░░░░░░░░░░░██
+ //                       ██░░░░░░░░░░░░░░░░██████░░░░░░░░░░░░░░░░██
+ //                     ██░░░░░░░░░░░░░░░░░░██████░░░░░░░░░░░░░░░░░░██
+ //                     ██░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░██
+ //                       ██████████████████████████████████████████
+ //
+ // If you happen to change the layout of this structure,
+ // this will lead to an AccessViolationException in netCore when using a more recent version of the nuget.
+ // If you need to modify the definition, create a new interface NativeCallTargetDefinition# that will be consumed by the native layer
```

Comment:
Is it really valid for OTel? Shouldn't it be changed to information that it should be keept in sync with `_CallTargetDefinition` on native code part?

For now, we do not support custom instrumentations.

Reply:
Updated in 4db1eea7102481f77ae27e1d7cf4e4ed2fbd13e7 . I revised the note to mention that it just needs to be kept up-to-date with `_CallTargetDefinition` native struct